pkg/k8s/watchers/pod.go

```
@@ -234,6 +245,18 @@ func (k *K8sWatcher) updateK8sPodV1(oldK8sPod, newK8sPod *slim_corev1.Pod) error
 	newPodLabels := newK8sPod.ObjectMeta.Labels
 	labelsChanged := !comparator.MapStringEquals(oldPodLabels, newPodLabels)
 
+	if oldK8sPod.Name == "lrp-pod" || newK8sPod.Name == "lrp-pod" {
+		dump := comparator.Compare(oldK8sPod, newK8sPod)
+		log.Info("pod DIFF2")
+		log.Info(dump)
+	}
+	// The relevant updates are : pod with valid IP(s) and label updates.
+	if (len(oldK8sPod.Status.PodIP) == 0 && len(newK8sPod.Status.PodIP) > 0) ||
+		(len(oldK8sPod.Status.PodIPs) == 0 && len(newK8sPod.Status.PodIPs) > 0) ||
```

I was wondering if it's possible for this to silently ignore some pod updates if a pod's IPs changed. Did you consider this? Even if it's never done today and we don't want to handle that case, I wonder if it'd be worthwhile to pick up on that case & log a warning to at least signal when it happens, so if this ever does become common then we will get a clear signal in the logs rather than just waiting for a report of weird behaviour and have to figure out the case.

---

Pod IPs can change.